crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -414,7 +479,8 @@ where
                     let backend = cache.backend;
                     let free_offsets = cache.cache_free_offsets;
                     stored_pieces.extend(cache.cache_stored_pieces.into_iter());
-                    dangling_free_offsets.extend(free_offsets.into_iter());
+                    dangling_free_offsets
+                        .extend(backend.dangling_free_offsets(free_offsets).into_iter());
```

Comment:
@nazar-pc Here we extract all free_offsets that are before last_used_offset (they are dangling_offsets)